code/modules/overmap/ship_weaponry/weaponry/light_coilgun.dm

```diff
@@ -0,0 +1,38 @@
+/obj/machinery/ship_weapon/coilgun/light
+	name = "M302A7 piledriver light coilgun"
+	desc = "A miniaturized version of the Solarian Navy’s coilgun platform, the M302A7 is the latest version of a weapon class that has served with honor since the Interstellar War. A semiautomatic, magazine-loaded design, it excels at applying pressure to enemy shields and hulls."
+	desc_extended = "Pressed for a smaller version of a coilgun to fit onto escort craft, the Alliance contracted what was then Necropolis Industries to design and manufacture the Navy’s first coilgun for corvette use in the mid-2200s. The M2307 is a post-Interstellar War modernization of earlier models that had been steadily improved upon over time. Though phased out of use in some battlegroups in favor of the M455A1 light coilgun, it remains a common sight in many fleets."
+	icon = 'icons/obj/machinery/ship_guns/sol_light_coilgun.dmi'
+	icon_state = "weapon_base"
+	max_ammo = 4
```

Comment:
My thought for the balance is to lower the maximum ammo from 4 to 2. I'd view the identity of coilguns as similar to the large single-shot cannons such as the lammergeier or longbow - it has extremely lethal projectiles, but low ammo capacity, so you need to make your shots count. High ammo capacities should be exclusive to automatic chemical guns like the grauwolf and francisca.